cla-backend-go/signatures/repository.go

```diff
@@ -192,6 +217,71 @@ func (repo repository) getSignatureCount(tableName string, filter expression.Con
 	return count, nil
 }
 
+// getSignatureManagerCount returns the total number of CLA Managers and unique CLA Managers in the system
+func (repo repository) getSignatureManagerCounts(tableName string, filter expression.ConditionBuilder) (int64, int64, error) {
+	var count int64
+	managerMap := make(map[string]bool)
+
+	// Use the nice builder to create the expression
+	expr, err := expression.NewBuilder().WithProjection(buildManagerProjection()).WithFilter(filter).Build()
+	if err != nil {
+		log.Warnf("error building expression for signature scan, error: %v", err)
+		return 0, 0, err
+	}
+
+	// Assemble the query input parameters
+	scanInput := &dynamodb.ScanInput{
+		ExpressionAttributeNames:  expr.Names(),
+		ExpressionAttributeValues: expr.Values(),
+		FilterExpression:          expr.Filter(),
+		ProjectionExpression:      expr.Projection(),
+		TableName:                 aws.String(tableName),
+	}
+
+	var lastEvaluatedKey string
+
+	// Loop until we have all the records
+	for ok := true; ok; ok = lastEvaluatedKey != "" {
+		results, errQuery := repo.dynamoDBClient.Scan(scanInput)
+		if errQuery != nil {
+			log.Warnf("error retrieving signatures for CLA Manager Count, error: %v", errQuery)
+			return 0, 0, errQuery
+		}
+
+		var dbManagersModel []DBManagersModel
+		err := dynamodbattribute.UnmarshalListOfMaps(results.Items, &dbManagersModel)
+		if err != nil {
+			log.Warnf("error unmarshalling signatures from database, error: %v", err)
+			return 0, 0, err
+		}
+
+		for _, record := range dbManagersModel {
+			if record.SignatureACL != nil {
+				// Tally the total count
+				count += int64(len(record.SignatureACL))
+				// Add the individual managers to our map - duplicate entries have the same key
+				for _, manager := range record.SignatureACL {
+					managerMap[manager] = true
+				}
+			}
+		}
+
+		if results.LastEvaluatedKey["signature_id"] != nil {
```

Comment:
should we use it like 
if len(results.LastEvaluatedKey) > 0 {
scanInput.ExclusiveStartKey = results.LastEvaluatedKey
} else {
break
}

Reply:
Actually, I think the LastEvaluatedKey will be the `signature_id` column only, since it is the key for this table.  So, this approach is precise. However, we must ensure the key is returned as part of the projections/response. We use this pattern throughout the entire codebase.